model/voucher-code.js

```diff
@@ -0,0 +1,45 @@
+const mongoose = require('mongoose');
+const bcyrpt = require('bcrypt');
+const Schema = mongoose.Schema;
+
+const voucher = Schema({
+    code : {type : String, required: true},
+    recepient : {type : Schema.Types.ObjectId,  ref: 'Recepient'},
+    offer : {type : Schema.Types.ObjectId,  ref: 'Offer'},
+    used : {type: Boolean, default: false},
+    track : {type: Date},
+    expiryDate: {type: Date}
+    
+})
+
+voucher.pre("save", function(next){
+    const mycode = this;
```

Comment:
use a good naming that describe what you're doing, you can also add some comment